vispy/app/backends/_wx.py

```diff
@@ -214,38 +214,44 @@ def __init__(self, *args, **kwargs):
         else:
             self._gl_context = context.shared.ref._gl_context
         
-        style = (wx.MINIMIZE_BOX | wx.MAXIMIZE_BOX | wx.CLOSE_BOX |
-                 wx.SYSTEM_MENU | wx.CAPTION | wx.CLIP_CHILDREN)
-        style |= wx.NO_BORDER if not dec else wx.RESIZE_BORDER
-        self._init = False
-        Frame.__init__(self, parent, wx.ID_ANY, title, position, size, style)
-        if not resize:
-            self.SetSizeHints(size[0], size[1], size[0], size[1])
-        if fs is not False:
-            if fs is not True:
-                logger.warning('Cannot specify monitor number for wx '
-                               'fullscreen, using default')
-            self._fullscreen = True
+        if parent is None:
+            style = (wx.MINIMIZE_BOX | wx.MAXIMIZE_BOX | wx.CLOSE_BOX |
+                     wx.SYSTEM_MENU | wx.CAPTION | wx.CLIP_CHILDREN)
+            style |= wx.NO_BORDER if not dec else wx.RESIZE_BORDER
+            self._frame = Frame(None, wx.ID_ANY, title, position, size, style)
+            if not resize:
+                self._frame.SetSizeHints(size[0], size[1], size[0], size[1])
+            if fs is not False:
+                if fs is not True:
+                    logger.warning('Cannot specify monitor number for wx '
+                                   'fullscreen, using default')
+                self._fullscreen = True
+            else:
+                self._fullscreen = False
+            _wx_app.SetTopWindow(self._frame)
+            parent = self._frame
+            self._frame.Raise()
+            self._frame.Bind(wx.EVT_CLOSE, self.on_close)
         else:
+            self._frame = None
             self._fullscreen = False
-        _wx_app.SetTopWindow(self)
-        
-        self._canvas = glcanvas.GLCanvas(self, wx.ID_ANY, wx.DefaultPosition,
-                                         wx.DefaultSize, 0, 'GLCanvas',
-                                         self._gl_attribs)
+        self._init = False
+        glcanvas.GLCanvas.__init__(self, parent, wx.ID_ANY, pos=position,
+                                   size=size, style=0, name='GLCanvas',
+                                   attribList=self._gl_attribs)
+            
+        self._canvas = self
```

Comment:
I assume this is just for testing, so this is just a reminder that eventually we should get rid of this and just do `self.` instead of `self._canvas.`